app/src/main/java/org/sil/storyproducer/controller/phase/PhaseBaseActivity.kt

```diff
@@ -67,7 +67,12 @@ abstract class PhaseBaseActivity : BaseActivity(), AdapterView.OnItemSelectedLis
         super.onPause()
```

Comment:
Hey Bryan, the onPause() event is called anytime a user is leaving an activity. "It indicates that the activity is no longer in the foreground" ([docs](https://developer.android.com/guide/components/activities/activity-lifecycle#onpause)). The purpose of the below code is to save the story state in case the user is leaving the app completely. Unfortunately since the user was leaving the activity when they switched workspaces, this onPause() method would still get called. This caused the rouge story to appear in the new workspace instead of saving the old one.

Reply:
Hi @noneel, I would think that the story state should be saved **before** the workspace switch is executed? 
Is it supposed to be saved before via onPause()? or perhaps it is already saved another way? 
If via onPause() - is the 'leaving the activity' triggered when the user selects a different workspace folder? Should it be triggered earlier when the story page is displayed and the user taps the main menu icon?

Reply:
Yes, the save is currently in the PhaseBaseActivity, which means that it will _Save_ once any phase runs the onPause() method because _almost_ all Phases are derived from PhaseBaseActivity. This happens **before** the user switches workspaces because I believe it will run when they click on the menu button since it will bring in a new activity. We could move this to the story page, however, that might make it so the user would have to go to the story page to _Save_ the current workspace. I guess at this point it is more of a design decision about when the workspace actually saved.

Reply:
Hi @noneel , It appears that jumpToPhase() (line 225~) starts an Android 'Activity' every time a phase is displayed. Then OnPause() saves the story json every time some other Activity comes to the foreground. Sounds good so far. Can we determine definitively if clicking the hamburger menu starts a new Activity, or does clicking the menu item 'Select SP Templates folder' start a new Activity?  I assume that the start of one of those new Activities triggers the individual phase activity's OnPause()? 
It bugs me that the individual phase activity's OnPause() is apparently being called **after** the Workspace folder has been changed (which takes time - it requires the user to select the folder). Or perhaps onPause() is called a second time unnecessarily? Or perhaps the Thread doing the story.toJson was delayed? 
Would it be informative to use the debugger to watch when PhaseBaseActivity's onPause() is called during different scenarios? We could also watch the file time attribute on the story.json file to watch when it is saved during different scenarios, but it's tricky to make the file manager refresh its display.